spec/requests/api/provision_requests_spec.rb

```diff
@@ -95,6 +95,146 @@
     end
   end
 
+  context "AWS advanced provision requests" do
+    let(:aws_dialog) do
+      dialog = FactoryGirl.create(:miq_dialog_aws_provision)
+      allow_any_instance_of(MiqRequestWorkflow).to receive(:dialog_name_from_automate).and_return(dialog.name)
+    end
+    let(:ems) { FactoryGirl.create(:ems_amazon_with_authentication) }
+    let(:template) do
+      FactoryGirl.create(:template_amazon, :name => "template1", :ext_management_system => ems)
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon, :ems_id => ems.id) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
+    let(:cloud_subnet1) do
+      FactoryGirl.create(:cloud_subnet, :ems_id => ems.id, :cloud_network => cloud_network1, :availability_zone => az)
+    end
+    let(:security_group1) { FactoryGirl.create(:security_group, :ext_management_system => ems) }
+    let(:floating_ip1) { FactoryGirl.create(:floating_ip_amazon, :address => "1.1.1.1") }
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {
+          "owner_first_name" => api_config(:user_name),
+          "owner_last_name"  => api_config(:user_name),
+          "owner_email"      => "user@example.com"
+        }
+      }
+    end
+
+    let(:provreq1_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          #"security_groups"             => security_group1.id,
```

Comment:
not sure why, sg and floating_ips do not work anymore, investigating.
the rest works, thanks a lot @gmcculloug, must have had my head too much in it to think straight